### ISSUE TITLE
Adldap::search()->users()->get() returns empty result

### ISSUE THREAD

**Issue by chrisribal** [Adldap2/Adldap2-Laravel#619]
- Laravel Version: 5.7.13
- Adldap2-Laravel Version: 5.0
- PHP Version: 7.1.20-1+ubuntu16.04.1+deb.sury.org+1
- LDAP Type: OpenLDAP

When trying to get the list of all users from my LDAP server, Adldap2 returns an empty array. After some research i found out, that the filter function `search()->users()` is equal to:

    $ldap->search()->where([
        ['objectclass', '=', 'inetorgperson'],
        ['objectclass', '=', 'person'],
        ['objectclass', '!', 'contact']
    ])

If i remove the last condition `['objectclass', '!', 'contact']` or change the operator to `!=` the script returns the correct list of my ldap users. What's the difference between these operators?

Of course, none of my ldap users has the class 'contact' assigned to it.

**Comment by stevebauman** [comment]
Hi @chrisribal, these two operators are supposed to operate the same, the `!=` operator is just an alias for the `!` operator to allow developers to use similar operators as they're used to in SQL.

That's strange that you're encountering that issue though, let me look into this.

I did also notice you're using OpenLDAP, did you change your `schema` setting in your `ldap.php` file to `Adldap\Schemas\OpenLDAP::class`?

**Comment by stevebauman** [comment]
Unable to reproduce this. Please comment back if you still encounter this using the `OpenLDAP` schema.

**Comment by ThomHurks** [comment]
@stevebauman I'm also having this issue. ```(!(objectclass=contact))``` is suddenly added to my search queries, breaking authentication.

**Comment by chrisribal** [comment]
I also ensured, that i've set the schema to OpenLDAP. I tried to directly query my LDAP via CLI (`ldapsearch`) and Apache Directory Studio, without any use of ADLDAP2. And i could reproduce that issue. It seems that the OpenLDAP Server doesn't know how to deal with `(!(objectclass=contact))`, although according to all LDAP documentations this filter value seems fine.

So maybe this is more of a general LDAP Server issue?
